extension/oidcextension/authenticator.go

```diff
@@ -29,47 +29,51 @@ import (
 
 	"github.com/coreos/go-oidc"
 	"google.golang.org/grpc"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
 )
 
 type oidcAuthenticator struct {
 	attribute string
-	config    OIDC
+	config    Config
 	provider  *oidc.Provider
 	verifier  *oidc.IDTokenVerifier
 
-	unaryInterceptor  unaryInterceptorFunc
-	streamInterceptor streamInterceptorFunc
+	unaryInterceptor  configauth.UnaryInterceptorFunc
+	streamInterceptor configauth.StreamInterceptorFunc
+	host              component.Host
 }
 
 var (
-	_ Authenticator = (*oidcAuthenticator)(nil)
-
-	errNoClientIDProvided                = errors.New("no ClientID provided for the OIDC configuration")
-	errNoIssuerURL                       = errors.New("no IssuerURL provided for the OIDC configuration")
+	errNoAudienceProvided                = errors.New("no Audience provided for the Config configuration")
```

Comment:
Could you leave this change as part of another PR? I'm not sure I agree with this change, especially because our underlying library uses this nomenclature.

Reply:
I was mostly put off by the change in nomenclature between this error and the config for the extension, since that looks for an `Audience` field while this error will talk about the `ClientID`